src/contacts/model/ContactModel.ts

```diff
@@ -32,13 +34,18 @@ export class ContactModel {
 	}
 
 	async getLoadedContactListInfos(): Promise<ReadonlyArray<ContactListInfo>> {
-		await this.loadContactLists()
-		return this.contactListInfo
+		// prevent re-loading them when we already have them
+		// this is not perfect and might still start loads in parallel
+		if (this.contactListInfo() === undefined) {
+			await this.loadContactLists()
+		}
+		return this.contactListInfo()
 	}
 
 	/** might be empty if not loaded yet */
-	getContactListInfos(): ReadonlyArray<ContactListInfo> {
-		return this.contactListInfo
+	getContactListInfos(): Stream<ReadonlyArray<ContactListInfo>> {
+		// defensive clone
+		return this.contactListInfo.map(identity)
```

Comment:
is this different from `.slice()`?

Reply:
yes, it does not return a copy of an array